tests/test/unify.rs

```diff
@@ -154,7 +154,7 @@ fn equality_binder() {
                 }
             }
         } yields {
-            "Unique; for<?U1> { \
+            "Unique; for<?U0> { \
```

Comment:
Why...did this change...?

Reply:
It looks like this was actually running into the ucanonicalization bug fixed in #529, with that fix applied the output for this test is unchanged. I've reverted this change in the latest commits.